### PR TITLE
Edgecloud 5229

### PR DESCRIPTION
**Pull request by ah1053**
Impl. of this document https://mobiledgex.atlassian.net/wiki/spaces/SWDEV/pages/2391867393/EdgeEvents+Developer+Usage

**EdgeEventsManager** (The UI component responsible for starting and stopping EdgeEvents)
1. Replaced NewFindCloudletHandler with OnConnectionUpgrade, OnConnectionFailure and EdgeEvents Exceptions.
2. Updated encapsulation for EdgeEventsManager methods.
3. Moved Methods that don't need to run on UI thread or can be stateless to EdgeEventsUtil (ValidateConfig, GetLatencyTestPort).
4. Used static references to Enums to avoid repetition (ex. EdgeEventsError.errorName).
5. Moved Public methods and vars next to each other and added descriptions for public methods and vars.

**EdgeEventsConfig** 

6. Removed FindCloudletEvent and EdgeEventsStatus as they are irrelevant in the new approach.

**MobiledgeXIntegration**

7. Based on https://github.com/mobiledgex/edge-cloud-sdk-unity/pull/206#discussion_r816907274 Added specific exception for checking whether OnConnectionUpgrade, OnConnectionFailure are assigned or not.

**Code Examples**

8. Update code Examples (ExampleREST, ExampleUDP, ExampleWebSocket, ExampleNonCellular)

### CODE REVIEW

File: Runtime/Scripts/EdgeEventsUtil.cs [hunk 1]
```diff
@@ -369,5 +381,121 @@ public static IEnumerator RespondToLatencyRequest(ConnectionDetails connectionDe
...
+      catch (AppPortException)
+      {
+        appPort = null;
+      }
+
+      if (appPort == null)
+      {
+        if (proto == LProto.Tcp)
+        {
+          Logger.LogWarning("No TCP ports exists on your App, It's recommended to use TCP ports for latency testing as Connect Tests is more reliable than Ping Tests.");
```
Comment (brucearmstrong) [inline Runtime/Scripts/EdgeEventsUtil.cs:485]: Nit: "No TCP ports exist in your App"

**Review by brucearmstrong — APPROVED** [review]
Lengthy, but looks to be pretty straightforward. LGTM.

File: Runtime/Scripts/EdgeEventsUtil.cs [hunk 2]
```diff
@@ -350,8 +362,8 @@ public static IEnumerator RespondToLatencyRequest(ConnectionDetails connectionDe
       {
         throw new NullReferenceException("edgeEventsConnection is null, Make sure you have a running edge events connection or that you are passing the correct arguments.");
       }
-      DeviceInfoDynamic deviceInfoDynamic = EdgeEventsManager.GetDynamicInfo(connectionDetails.matchingEngine);
-      AppPort appPort = EdgeEventsManager.GetLatencyTestPort(connectionDetails.mobiledgexManager, latencyPort);
+      DeviceInfoDynamic deviceInfoDynamic = EdgeEventsManager.GetDeviceDynamicInfo(connectionDetails.matchingEngine);
```
Comment (lgarner) [inline Runtime/Scripts/EdgeEventsUtil.cs:365]: DeviceInfoDynamic ?

File: Runtime/Scripts/ExampleWebSocket.cs [hunk 3]
```diff
@@ -23,7 +23,8 @@ IEnumerator Start()
   async void GetEdgeConnection()
   {
     mxi = new MobiledgeXIntegration(FindObjectOfType<EdgeEventsManager>());
-    mxi.NewFindCloudletHandler += HandleFindCloudlet;
+    mxi.OnConnectionFailure = OnConnectionFailure;
```
Comment (lgarner) [inline Runtime/Scripts/ExampleWebSocket.cs:26]: Q: Not += ?

**Review by lgarner — APPROVED** [review]
👍

File: Runtime/Scripts/EdgeEventsUtil.cs (hunk 2 above)
Comment (ah1053) [reply Runtime/Scripts/EdgeEventsUtil.cs:365]: This is coming from C#
Screen Shot 2022-03-11 at 10 11 24 AM

**Pull request merged**